### DISCUSSION TITLE
Prettier Config

### DISCUSSION THREAD

**Discussion by eliotwrobson** [PrairieLearn/PrairieLearn#9443] [Dev]
The config files that set VSCode to format-on-save have been a big time saver for me. However, in cases where I'm editing an HTML file, prettier's formatting doesn't play nice with the `<pl-code>` element (for example), since that element relies on whitespace. Is there a way to configure prettier to ignore only this element?

**Comment by nwalters512** [comment]
Long story short: no. This is the price we pay for overloading HTML syntax with our own semantics!

**Reply by jonatanschroeder** [reply]
If your code is not dynamic, I strongly encourage you to save it in a separate file and use `source-file-name` instead.

**Reply by firasm** [reply]
It also doesn't play nice with the markdown tag 😭 - surely there **must** be a way to configure this. I've thought this multiple times, searched, and had no luck (or ran out of time)

**Reply by nwalters512** [reply]
`<markdown>` is a PrairieLearn-ism that is all but guaranteed to not have support in other tools.

**Reply by firasm** [reply]
I know 🙂, but until PR #7467 lands, it's the only way (I know) to write readable questions!

**Comment by jonatanschroeder** [comment]
I just tested, and you can precede it with `<!-- prettier-ignore -->`. You'd have to do it for every single `pl-code`, though.

**Reply by eliotwrobson** [reply]
I think that's still useful, a lot of the use cases are for questions that only have one or two pl-code elements.

**Reply by firasm** [reply]
Oooh - there is also [range-ignore](https://prettier.io/docs/en/ignore#range-ignore), you can also do this:

```
<!-- prettier-ignore-start -->
<!-- SOMETHING AUTO-GENERATED BY TOOLS - START -->

| MY | AWESOME | AUTO-GENERATED | TABLE |
|-|-|-|-|
| a | b | c | d |

<!-- SOMETHING AUTO-GENERATED BY TOOLS - END -->
<!-- prettier-ignore-end -->
```
